src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,32 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Cant save actor to DB. Actor: " + actor, e);
+        } finally {
+            transaction.commit();
+            session.close();
+        }
+        return actor;
     }
 
     @Override
     public Optional<Actor> get(Long id) {
-        return null;
+        try (Session session = factory.openSession();) {
+            Actor actor = session.get(Actor.class, id);
+            return Optional.ofNullable(actor);
```

Comment:
```suggestion
            return Optional.ofNullable(session.get(Actor.class, id));
```

Reply:
I recommend adding the same changes here and in the other "get" methods as well!